Code/GraphMol/MolDraw2D/MolDraw2DJS.cpp

```diff
@@ -0,0 +1,142 @@
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+//
+
+#ifdef __EMSCRIPTEN__
+
+#include <GraphMol/MolDraw2D/MolDraw2DJS.h>
+
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <Geometry/point.h>
+#ifdef RDK_BUILD_FREETYPE_SUPPORT
+#include <GraphMol/MolDraw2D/DrawTextFTJS.h>
+#endif
+#include <GraphMol/MolDraw2D/DrawTextJS.h>
+
+#include <boost/format.hpp>
+#include <sstream>
+#include <cmath>
+
+namespace RDKit {
+
+std::string DrawColourToSVG(const RDKit::DrawColour &col);
+
+// ****************************************************************************
+void MolDraw2DJS::initDrawing() {}
+
+// ****************************************************************************
+void MolDraw2DJS::initTextDrawer(bool noFreetype) {
+  double max_fnt_sz = drawOptions().maxFontSize;
+  double min_fnt_sz = drawOptions().minFontSize;
+
+  if (noFreetype) {
+    text_drawer_.reset(new DrawTextJS(max_fnt_sz, min_fnt_sz, d_context));
+  } else {
+#ifdef RDK_BUILD_FREETYPE_SUPPORT
+    try {
+      text_drawer_.reset(new DrawTextFTJS(max_fnt_sz, min_fnt_sz,
+                                          drawOptions().fontFile, d_context));
+    } catch (std::runtime_error &e) {
+      BOOST_LOG(rdWarningLog)
+          << e.what() << std::endl
+          << "Falling back to native JS text handling." << std::endl;
+      text_drawer_.reset(new DrawTextJS(max_fnt_sz, min_fnt_sz, d_context));
+    }
+#else
+    text_drawer_.reset(new DrawTextJS(max_fnt_sz, min_fnt_sz, d_context));
+#endif
+  }
+}
+
+// ****************************************************************************
+void MolDraw2DJS::drawLine(const Point2D &cds1, const Point2D &cds2) {
+  Point2D c1 = getDrawCoords(cds1);
+  Point2D c2 = getDrawCoords(cds2);
+  std::string col = DrawColourToSVG(colour());
+  unsigned int width = getDrawLineWidth();
+  std::string dashString = "";
+  const DashPattern &dashes = dash();
+
+  d_context.call<void>("beginPath");
+  d_context.set("lineWidth", width);
+  d_context.set("strokeStyle", col);
+  if (dashes.size()) {
+    d_context.call<void>("setLineDash", emscripten::typed_memory_view(
+                                            dashes.size(), dashes.data()));
+  }
+  d_context.call<void>("moveTo", std::round(c1.x), std::round(c1.y));
+  d_context.call<void>("lineTo", std::round(c2.x), std::round(c2.y));
+  d_context.call<void>("stroke");
+  if (dashes.size()) {
+    static const DashPattern nodash;
+    d_context.call<void>("setLineDash", emscripten::typed_memory_view(
+                                            nodash.size(), nodash.data()));
+  }
+}
+// ****************************************************************************
+void MolDraw2DJS::drawPolygon(const std::vector<Point2D> &cds) {
+  PRECONDITION(cds.size() >= 3, "must have at least three points");
+
+  std::string col = DrawColourToSVG(colour());
+  unsigned int width = getDrawLineWidth();
```

Comment:
In the PR I am about to submit, getDrawLineWidth() returns a double.

Reply:
That totally makes sense. Can fix this then (or together with your PR if we merge this one first)